eng/pipelines/templates/stages/archetype-cpp-release.yml

```diff
@@ -65,3 +65,28 @@ stages:
                           TargetLanguage: 'cpp'
                           # we override the regular script path because we have cloned the build tools repo as a separate artifact.
                           ScriptPath: '$(Pipeline.Workspace)/azure-sdk-build-tools/scripts/copy-docs-to-blobstorage.ps1'
+
+          - ${{if ne(artifact.options.skipUpdatePackageVersion, 'true')}}:
+            - deployment: UpdatePackageVersion
+              displayName: "Update Package Version"
+              condition: and(succeeded(), ne(variables['Skip.UpdatePackageVersion'], 'true'))
+              environment: github
+
+              pool:
+                vmImage: windows-2019
+
+              strategy:
+                runOnce:
+                  deploy:
+                    steps:
+                      - checkout: self
+                      - pwsh: |
+                          eng/Update-PkgVersion.ps1 -ServiceDirectory '${{parameters.ServiceDirectory}}' -PackageName '${{artifact.name}}'
+                        displayName: Increment package version
+                      - template: tools/clone-buildtools/clone-buildtools.yml@azure-sdk-build-tools
```

Comment:
We should start using the templates in eng/common for this and other things in this file now that @chidozieononiwu has ported them there.

Reply:
Created #56 to track this across the repo. It looks like Clone-BuildTools itself is _not_ yet in eng/common.